src/syscall_table/mod.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+mod aarch64;
+mod x86_64;
+
+use crate::backend::TargetArch;
+use std::collections::HashMap;
+
+/// Creates and owns a mapping from the arch-specific syscall name to the right number.
+#[derive(Debug)]
+pub(crate) struct SyscallTable {
+    map: HashMap<String, i64>,
+    arch: TargetArch,
+}
+
+/// Rough number of syscalls.
+const MAP_CAPACITY: usize = 352;
+
+impl SyscallTable {
+    pub fn new(arch: TargetArch) -> Self {
+        let mut instance = Self {
```

Comment:
Instead of creating an instance with capacity, we should just directly initialize self without using the `populate_map` function. This is possible once we return a hashmap from `make_syscall_table`.